lib/expression/v2.ex

```diff
@@ -174,11 +167,36 @@ defmodule Expression.V2 do
   """
   @spec eval_as_string(String.t(), Context.t()) :: String.t()
   def eval_as_string(expression, context \\ Context.new()) do
-    eval(expression, context)
-    |> Enum.map(&default_value(&1, context))
-    |> Enum.map_join("", &stringify/1)
+    {:ok, ast} = parse(expression)
+
+    ast
+    |> eval_ast(context)
+    |> Enum.zip(ast)
+    |> Enum.map_join("", fn
+      {nil, [{"__property__", _parts} = property]} ->
+        "@" <> unwrap_property(property)
+
+      {value, _ast} ->
+        value
+        |> default_value(context)
+        |> stringify()
+    end)
```

Comment:
this is what the change is about, the existing tests cover this and confirm that it's still working as expected.